packages/core/src/domain/session/sessionStoreManager.ts

```diff
@@ -0,0 +1,256 @@
+import { isSessionInExpiredState } from './sessionStore'
+
+export interface SessionStoreManager {
+  expandOrRenewSession: () => void
+  expandSession: () => void
+  getSession: () => SessionState
+  renewObservable: Observable<void>
+  expireObservable: Observable<void>
+  expire: () => void
+  stop: () => void
+}
+
+/**
+ * Different session concepts:
+ * - tracked, the session has an id and is updated along the user navigation
+ * - not tracked, the session does not have an id but it is updated along the user navigation
+ * - inactive, no session in store or session expired, waiting for a renew session
+ */
+export function startSessionStoreManager<TrackingType extends string>(
+  options: StoreInitOptions,
+  productKey: string,
+  computeSessionState: (rawTrackingType?: string) => { trackingType: TrackingType; isTracked: boolean }
+): SessionStoreManager {
+  const renewObservable = new Observable<void>()
+  const expireObservable = new Observable<void>()
+
+  const sessionStore = initCookieStore(options)
+  const { clearSession, retrieveSession, storeAccessOptions } = sessionStore
+
+  const watchSessionTimeoutId = setInterval(watchSession, storeAccessOptions.pollDelay)
+  let sessionCache: SessionState = retrieveActiveSession()
+
+  function expandOrRenewSession() {
+    let isTracked: boolean
+    processSessionStoreOperations(
+      {
+        process: (sessionState) => {
+          const synchronizedSession = synchronizeSession(sessionState)
+          isTracked = expandOrRenewCookie(synchronizedSession)
+          return synchronizedSession
+        },
+        after: (sessionState) => {
+          if (isTracked && !hasSessionInCache()) {
+            renewSessionInCache(sessionState)
+          }
+          sessionCache = sessionState
+        },
+      },
+      sessionStore
+    )
+  }
+
+  function expandSession() {
+    processSessionStoreOperations(
+      {
+        process: (sessionState) => (hasSessionInCache() ? synchronizeSession(sessionState) : undefined),
+      },
+      sessionStore
+    )
+  }
+
+  /**
+   * allows two behaviors:
+   * - if the session is active, synchronize the session cache without updating the session store
+   * - if the session is not active, clear the session store and expire the session cache
+   */
+  function watchSession() {
+    processSessionStoreOperations(
+      {
+        process: (sessionState) => (!isActiveSession(sessionState) ? {} : undefined),
+        after: synchronizeSession,
+      },
+      sessionStore
+    )
+  }
+
+  function synchronizeSession(sessionState: SessionState) {
+    if (!isActiveSession(sessionState)) {
+      sessionState = {}
+    }
+    if (hasSessionInCache()) {
+      if (isSessionInCacheOutdated(sessionState)) {
+        expireSessionInCache()
+      } else {
+        sessionCache = sessionState
+      }
+    }
+    return sessionState
+  }
+
+  function expandOrRenewCookie(sessionState: SessionState) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏we should probably update this function name to not be "cookie-centric" 🙂

Reply:
Renamed to `expandOrRenewSessionState`